controllers/argocd/policyrule.go

```diff
@@ -360,3 +363,18 @@ func appendOpenShiftNonRootSCC(rules []v1.PolicyRule, client client.Client) []v1
 	}
 	return rules
 }
+
+func appendBatchPolicyRule(rules []v1.PolicyRule) []v1.PolicyRule {
+	batchPolicyRule := v1.PolicyRule{
+		APIGroups: []string{
+			"batch",
+		},
+		Resources: []string{
+			"jobs",
+		},
+		Verbs: []string{
+			"create",
+		},
+	}
+	return append(rules, batchPolicyRule)
+}
```

Comment:
I don't think a new function is required for this purpose, we could probably just add this set of permissions as an additional array element in the existing ruleset for the server role and clusterrole directly

Reply:
Yeah, appending it inline would be more readable. I've made the change to inline.